packages/library-base/src/commonMain/kotlin/io/realm/kotlin/types/Distance.kt

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.kotlin.types
+
+/**
+ * This class represents an equatorial distance that can be used in geospatial queries like those
+ * represented by a [GeoCircle].
+ */
+public data class Distance private constructor(
```

Comment:
Yes, it is the distance on the earth. I guess we could have also gone with `GeoDistance`, but `Distance` was the name used in the technical scope: https://docs.google.com/document/d/1Z7CK79CqebEtVllVHTNEq8neKhoMWeFCDjKBrn_pq1Y/ and I don't think that is particular confusing? And it is shorter than `GeoDistance`.

Reply:
It confuses me because it mixed linear and angular distances.

We implicitly use the earth's radius to correlate between them, but we are not explicitly stating in the documentation that we are using it. Instead, we could be explicit:

```kotlin
val distance: Distance = 1.2.earthRadians
val distance: Distance = 1.2.earthDegrees
val distance: Distance = 10.4.km
val distance: Distance = 10.5.miles
```

Reply:
Maybe I'm missing something obvious here, but how could `radians` be interpreted in any other way than Earth radians? Like what can it potentially be confused with?

Reply:
I also had the same feeling like Clemente. It isn't at all obvious that radians are "earth radians" (if such a term even exists!). But once you start interpreting this as "equatorial distance in radians" it starts to be ok to work with. I think it is just something people will need to grasp and then it is ok. I tried to suggest updates to some docs by adding explicit "equatorial distance" here and there to speed up the "learning curve" to make it more clear. Anyway the `km` should make it possible to work with it without bothering with the internals.

Reply:
The whole Geo-Spatial solution is generic, it can work on any sphere. It is this `Distance` implementation that tights it to the Earth. 

The thing is that this solution works for other spheres too, planets or simulations. If these KM/Miles are used on a different sphere than the earth it would not be correct. Adding a comment about it would be fine.

[Here](https://github.com/realm/realm-kotlin/pull/1403/files/df78a0f1d9ac699c7edcea263c02e6f4849c714a#r1244320494) I tried to convert values Distance as a linear distance, that's why I wrote "earthRadians" (radians*earth radius = linear distance ). But we actually require angular distance so forget about it.